raster-foundry/raster-foundry
# Ensure correct footprints and tasks for pre-COGs

## Overview

Sometimes we need to COGify images on our own, but other times, they're _pre-COGs_:

![](https://static.wikia.nocookie.net/minority-report/images/e/e7/Precog_statue.jpg/revision/latest?cb=20170804105554)

When we have COGs already, we don't need to do work to turn the image into a COG. However, we _do_ need to do work to make sure some metadata is correct. In particular, for EPSG:4326 COGs, we were doing a bad job at task side length calculation, and then when I fixed that, we were doing a bad job at footprint calculation.

This PR makes it so that:

- we use a nice Web Mercator VRT for figuring out the task side length
- we bail on the complicated footprint calculation and just let the API server handle it
- we make sure that the scene is ready before we try to use its metadata in the database to calculate annotation project footprints

### Checklist


### Demo

For my local copy of a poison tiff:

```bash
$ time ./scripts/process-upload 3c5df351-5868-4fbb-883b-48a96ce32cf4
...
________________________________________________________
Executed in   20.92 secs   fish           external
   usr time  1108.50 millis  170.00 micros  1108.33 millis
   sys time  248.30 millis  119.00 micros  248.18 millis
```

Here's the [log stream](https://console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:log-groups/log-group/$252Faws$252Fbatch$252Fjob/log-events/jobStagingProcessUpload$252Fdefault$252Fd6e48a421de74e5d91e594915a45cbcd) for trying to process the same tif in staging. You can see that 24 minutes in it dies with some kind of SQL error.

## Testing Instructions

- Upload the poison tif from the linked issue in staging and wait for the job to die
- build the batch container `docker-compose build batch`
- make sure your servers are up-to-date just as a precaution `./sbt ';api/assembly;backsplash-server/assembly;batch/assembly'`
- bring up servers `./scripts/server`
- bring up GroundWork pointed to local servers
- upload the same poision tif
- process it `./scripts/process-upload <your id>`
- note that it doesn't take 24 minutes and die, but instead takes a short amount of time, succeeds, and you eventually get a footprint matching the imagery 🥖

Closes azavea/raster-foundry-platform#1328
